packages/api-types/src/datasets.ts

```diff
@@ -41,11 +41,16 @@ export enum DatasetStatus {
   Error = 'error',
 }
 
+export type DatasetCustomTypes = 'points' | 'lines' | 'geometries'
 export interface DatasetConfiguration {
   index?: string
   filePath?: string
-  propertyToInclude?: string
   srid?: number
+  file?: string
```

Comment:
Defining [optional properties for our interfaces is not a good habit in typescript](https://startup-cto.net/10-bad-typescript-habits-to-break-this-year/#6optionalproperties). In general this is a sign that the interface could still be extended in several data types that are more explicit and will provide us type checking at compile-time instead of run-time.
In this specific case maybe we should define `PointsDatasetConfiguration`, `LinesDatasetConfiguration` and `GeometriesDatasetConfiguration`, each one containing only the properties that are needed.

Reply:
Thanks for sharing! I'll start by the basic of having `DatasetConfiguration` and `EnviromentalDatasetConfiguration` and then lets discuss how to split them by other types.

I also created a new `type AnyDatasetConfiguration = DatasetConfiguration | EnvDatasetConfiguration` to update the dataset configuration type to use:
```js
configuration: AnyDatasetConfiguration | null
```
does it makes sense?

Reply:
Thanks @j8seangel! Great approach 💯 !